contracts/FlowCallbackScheduler.cdc

```diff
@@ -348,36 +368,26 @@ access(all) contract FlowCallbackScheduler {
             )
         }
 
-        // The estimate function calculates the required fee in Flow and expected execution time for 
-        // a callback based on timestamp, priority, and execution effort. 
+        /// The estimate function calculates the required fee in Flow and expected execution time for 
+        /// a callback based on timestamp, priority, and execution effort. 
         //
-        // If the provided arguments are invalid or the callback cannot be scheduled (e.g., due to 
-        // insufficient computation effort or unavailable time slots) the estimate function returns `nil`.
-        //        
-        // This helps developers ensure sufficient funding and preview the expected scheduling window, 
-        // reducing the risk of unnecessary cancellations.
+        /// If the provided arguments are invalid or the callback cannot be scheduled (e.g., due to 
+        /// insufficient computation effort or unavailable time slots) the estimate function returns `nil`.
+        ///        
+        /// This helps developers ensure sufficient funding and preview the expected scheduling window, 
+        /// reducing the risk of unnecessary cancellations.
         access(all) view fun estimate(
             data: AnyStruct?,
             timestamp: UFix64,
             priority: Priority,
             executionEffort: UInt64
         ): EstimatedCallback? {
-            // low priority callbacks are not supported
+            /// low priority callbacks are not supported
             if priority == Priority.Low {
                 return EstimatedCallback(flowFee: nil, timestamp: nil, error: "Invalid priority: low priority callbacks estimation not supported")
             }
 
-            if timestamp <= getCurrentBlock().timestamp {
-                return EstimatedCallback(flowFee: nil, timestamp: nil, error: "Invalid timestamp: timestamp is in the past")
-            }
-
-            if executionEffort > self.priorityEffortLimit[priority]! {
-                return EstimatedCallback(flowFee: nil, timestamp: nil, error: "Invalid execution effort: greater than available effort for priority")
-            }
-
-            if executionEffort < self.minimumExecutionEffort {
-                return EstimatedCallback(flowFee: nil, timestamp: nil, error: "Invalid execution effort: less than minimum execution effort")
-            }
+            self.checkScheduleArgs(timestamp: timestamp, priority: priority, executionEffort: executionEffort)
```

Comment:
I believe this means estimate would panic in case of invalid args which I don't think is good idea, since estimate might be used to test args and might fail whole transaction. I intentionally didn't want to panic, but let me know your thinking.